constants.go

```diff
@@ -0,0 +1,15 @@
+package paging
+
+// pagination system type
+const (
```

Comment:
Maybe "better" to name constants like this:

* `TypeOffset`
* `TypeCursor`
* `CursorModeDate`
* `CursorModeID`

IMHO, cleaner naming and much more easier for completion (If I type `Type`, completion will show me all Type constants).

That's purely subjective. My peanuts :)